src/main/java/org/opentripplanner/transit/raptor/util/LuceneBitSetIterator.java

```diff
@@ -0,0 +1,32 @@
+package org.opentripplanner.transit.raptor.util;
+
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.util.BitSet;
+import org.opentripplanner.transit.raptor.api.transit.IntIterator;
+
+/**
+ * TODO TGR
+ */
+public final class LuceneBitSetIterator implements IntIterator {
+
+  private final BitSet set;
+  private int nextIndex;
+
+  public LuceneBitSetIterator(BitSet set) {
+    this.set = set;
+    this.nextIndex = set.nextSetBit(nextIndex);
+  }
+
+  @Override
+  public int next() {
+    int index = nextIndex;
+    nextIndex =
+      index == set.length() - 1 ? DocIdSetIterator.NO_MORE_DOCS : set.nextSetBit(index + 1);
```

Comment:
Could you check the performance with `SparseFixedBitSet`, In that case the `nextSetBit` could be faster, since it checks each 4096-bit block only if at least one value is set, and it knows directly in which 64-bit segment. the regular `FixedBitSet` needs to check each block. It might be that linear scanning is still faster, as most of the data is already in the cache.

Reply:
Seems to be better than `org.apache.lucene.util.FixedBitSet`, but still worse than `java.util.BitSet`

```
java.util.BitSet


Worker: 
 ==> mc_destination : [  464,  441,  440,  444 ] Avg: 447.3

Total:  
 ==> mc_destination : [  772,  705,  709,  711 ] Avg: 724.3


org.apache.lucene.util.FixedBitSet


Worker: 
 ==> mc_destination : [  480,  459,  465,  466 ] Avg: 467.5

Total:  
 ==> mc_destination : [  779,  741,  750,  750 ] Avg: 755.0


org.apache.lucene.util.SparseFixedBitSet


Worker: 
 ==> mc_destination : [  480,  447,  449,  452 ] Avg: 457.0

Total:  
 ==> mc_destination : [  788,  726,  722,  720 ] Avg: 739.0
```